src/qt/bitcoingui.cpp

```diff
@@ -1352,20 +1353,27 @@ void BitcoinGUI::showModalOverlay()
         modalOverlay->toggleVisibility();
 }
 
-static bool ThreadSafeMessageBox(BitcoinGUI* gui, const std::string& message, const std::string& caption, unsigned int style)
+static bool ThreadSafeMessageBox(BitcoinGUI* gui, const bilingual_str& message, const std::string& caption, unsigned int style)
 {
     bool modal = (style & CClientUIInterface::MODAL);
     // The SECURE flag has no effect in the Qt GUI.
     // bool secure = (style & CClientUIInterface::SECURE);
     style &= ~CClientUIInterface::SECURE;
     bool ret = false;
+
+    QString detailed_message;
+    if (message.original != message.translated) {
+        detailed_message = QObject::tr("Original message:") + "\n" + QString::fromStdString(message.original);
```

Comment:
what is the point of translations again, if we show the original message anyway?

Reply:
This translated title, `QObject::tr("Original message:")`, is intended for non-english interface.
The message itself is not translated, obviously.
That was an initial idea, but I'm open for changes ;)

Reply:
yes, what is the point of showing the translation *and* the original message. If the user understood the original message (in English), what is the point of showing the translation?

Reply:
Translation - for a user.
Original message - for googling and referencing.

Reply:
I see. I guess fine with me, but at least the code could mention that.

Reply:
Add some comments?

Reply:
jup

Reply:
Done.